ReadMe.md
# Event Calendar

https://store.kde.org/p/998901/

Plasmoid for a calendar+agenda with weather that syncs to Google Calendar.

## Screenshots

![](https://i.imgur.com/qdJ71sb.jpg)
![](https://i.imgur.com/Ow8UlFj.jpg)


## A) Install via KDE

1. Right Click Panel > Panel Options > Add Widgets
2. Get New Widgets > Download New Widgets
3. Search: Event Calendar
4. Install
5. Right Click your current calendar widget > Alternatives
6. Select Event Calendar

## B) Install via GitHub

```
git clone https://github.com/Zren/plasma-applet-eventcalendar.git eventcalendar
cd eventcalendar
sh ./install
```

To update, run the `sh ./update` script. It will run a `git pull` then reinstall the applet. Please note this script will restart plasmashell (so you don't have to relog)!

## C) Install via Package Manager

Some awesome users seemed to have packaged this applet under `plasma5-applets-eventcalendar`.

* Arch: https://aur.archlinux.org/packages/plasma5-applets-eventcalendar/
* Chakra: https://chakralinux.org/ccr/packages.php?ID=7656

(Old) There's also a russian who's patched the widget with russian translations. It's out of date though, and we now bundle russian translations with the rest.

* ABF: https://abf.rosalinux.ru/victorr2007/plasma5-applet-eventcalendar

## Update to GitHub master

If you're asked to test something, you can do so by installing the latest unreleased code.

Beforehand, uninstall the AUR version if you are running Arch (you can reinstall after testing).

Then install pen the Terminal and run the following commands. Please note the install script will restart plasmashell so that you don't have to relog.

```
sudo apt install git
git clone https://github.com/Zren/plasma-applet-eventcalendar.git eventcalendar
cd eventcalendar
sh ./install --restart
```

When you've finished testing, you may wish to reinstall the KDE Store or AUR version. First uninstall the widget with the following command, then reinstall your desired version of the widget.

```
sh ./uninstall
```

## Configure

1. Right click the Calendar > Event Calendar Settings > Google Calendar
2. Copy the Code and enter it at the given link. Keep the settings window open.
3. After the settings window says it's synched, click apply.
4. Go to the Weather Tab > Enter your city id for OpenWeatherMap. If their search can't find your city, try googling it with [site:openweathermap.org/city](https://www.google.ca/search?q=site%3Aopenweathermap.org%2Fcity+toronto).




Changelog.md
## v76 - May 3 2022

* Do not show calendar border by default anymore.
* Hide Google Calendar/Tasks headers when not logged in to not confuse
* Optionally open google events in web month view (Pull Request #251)
* Hide "This event was added from Goals" in event description (Pull Request #236)
* Support Plasma 5.24 kcms (Set Formats, Language, Adjust Date and Time), and use ellipsis unicode.
* Actually implement custom height/width in twocolumn mode (Issue #299)
* Update hidden Advanced config
* Remove manual LocaleInstaller which was needed for AUR installs.
* Update i18n scripts, and reuse a couple translations from l10n.kde.org
* Fix broken datetime format in Japanese translation by @gyu-don (Pull Request #246)
* Add Slovenian translation by @Ugowsky (Issue #255)
* Add Finnish translation by https://www.opendesktop.org/u/ilkant
* Updated pt_BR translations by @Bernardozomer (Pull Request #276)
* Add Hebrew translation by @omeritzics (Pull Request #305)
* Updated Dutch translation by @Vistaus (Pull Request #306)

## v75 - June 1 2021

* Show conference button for Google Meet and Zoom by @gaganpreet (Issue #63 and Pull Request #243)
* Fix parsing WeatherCanada AB/MB city list.
* Move the calendar name in the tooltip below the description like on Google Calendar.
* Complete German translations by @TimL20 (Pull Request #237)
* Update Italian translations by @guidomazzone (Pull Request #234)
* Update Portuguese translations by @dmmleal (Pull Request #232)

## v74 - March 24 2021

* Make PlasmaNM optional using a QML Loader (Issue #212)
* Fix HTTP request code to fix weather not updating on connect when using OpenWeatherMap (Issue #218)
* Fix daily forecast in agenda when using WeatherCanada and add error handling.
* Show a notification 15min before events starts. Defaults with no sfx but can be enabled. (PR #221)
* Implement expiresTimeout in notification.py should we want notification to last longer in the future.
* Add event tooltip to agenda that always displays the calendar name, and the description if it's been hidden (Issue #152 and #195)
* Limit event description to 5 lines by default (Issue #196)
* Shift js files around. Use === instead of ==.
* Wrap hangout ToolButton in a Loader so we avoid creating unused controls (which loads an svg) when creating events.
* Add Portuguese european translations by @dmmleal (Pull Request #226 and #230)
* Updated Dutch translation by @Vistaus (Pull Request #228)

## v73 - January 18 2021

* Add a NetworkMonitor to detect when there's no internet (Issue #113)
* Automatically close HTTP 0 network not connected errors when reconnected (Issue #205)
* Fix timer sfx not working in certain locales (Issue #209)
* Update Russian translations by driglu4it on Element

## v72 - December 22 2020

* Only store the filename for enabled PlasmaCalendar plugins (Issue #204)
* Check if gcal items is undefined before reading length (Issue #203)
* Use 90% color step from bg to text for grid (Issue #201)
* icsjson.py fixes by @koosvriezen (Pull Request #184)
* Add Swedish translations by @EazyDefcon (Issue #197)
* Updated Italian translation by @guidomazzone (Pull Request #182)
* Updated Dutch translation by @Vistaus (Pull Request #184)

## v71 - November 13 2020

* Change the Google API key as we were hiting the daily quota due to 1 user still using v63. This requires you to re-login to Google Calendar Sync.
* Display alerts above the agenda when a widget update changes the API key. Also displays alerts when the web requests to google return with an error.
* Refactor the code of most config options. They old config values should migrate cleanly for the next few versions before the migration code is removed.
* Remove the final QtMultimedia use in ConfigSound to fix the config window crashing on Kubuntu and OpenSUSE (Issue #167)
* Allow double click in the Calendar to do nothing by @matty-r (Pull Request #170)
* Refactor Timer as QML Timer is untrustworthy on high refresh screens (Issue #129)
* Jump timer by 5min when duration over 15min or seconds when under 1min
* Disable the timer sfx control when showTimer=false (Pull Request #172)
* Can now configure the calendar's title date format by @matty-r (Pull Request #176)
* Use ColorScheme colors for GoogleCal login text. Also fix poorly colored date format preset buttons.
* Now uses a semi-transparent text color as the agenda item hover color when the color theme has low contrast. This fixes Oxygen's white on white when hovering.
* Merge reinstall code into the install script. Also add uninstall script.
* Add Italian translation by @guidomazzone (Issue #169 and Pull Request #179)

## v70 - October 14 2020

* Fix Save/Save/Cancel buttons in Edit Timer/Task/Event forms that were invisible in latest KDE Frameworks.
* Fix bug where toggling timer sfx in config doesn't apply change.
* Show CheckBox in NewEventForm when tasklist is selected. Gives a visual cue that the last event added was a task instead of an event.
* Vertical Align "to" text in DurationSelector.
* Fix EventItem height bug after closing EditForm.
* Add EditTaskForm based EditEventForm.
* Remove "notifications" dataengine to support Latte Dock (Issue #60). We also remove the QtMultimedia sfx that caused issues in (Issue #84). We now use notification.py for events as well. I've done my best to sanitize the event summary + body.
* Implement konsolekalendar add event with PIM calendars (Issue #43). Also added a konsolekalendar.py script to find the uid of an event.
* Check for err before parsing oauth token data. (Issue #147)
* Updated Dutch translations by @Vistaus (Pull Request #164)

## v69 - July 27 2020

* Add copy to clipboard in contextual actions by @navarroaxel (Pull Request #142) (Issue #28). Widget now requires Plasma 5.13.
* Swap Agenda and Calendar positions like Digital Clock will in Plasma 5.20. The Calendar appears above the Agenda when in single column mode.

package/contents/code/ColorIdMap.js
.pragma library

var colorIdMap = {
	"calendar": {
		"1": {
			"background": "#ac725e",
			"foreground": "#1d1d1d"
		},
		"2": {
			"background": "#d06b64",
			"foreground": "#1d1d1d"
		},
		"3": {
			"background": "#f83a22",
			"foreground": "#1d1d1d"
		},
		"4": {
			"background": "#fa573c",
			"foreground": "#1d1d1d"
		},
		"5": {
			"background": "#ff7537",
			"foreground": "#1d1d1d"
		},
		"6": {
			"background": "#ffad46",
			"foreground": "#1d1d1d"
		},
		"7": {
			"background": "#42d692",
			"foreground": "#1d1d1d"
		},
		"8": {
			"background": "#16a765",
			"foreground": "#1d1d1d"
		},
		"9": {
			"background": "#7bd148",
			"foreground": "#1d1d1d"
		},
		"10": {
			"background": "#b3dc6c",
			"foreground": "#1d1d1d"
		},
		"11": {
			"background": "#fbe983",
			"foreground": "#1d1d1d"
		},
		"12": {
			"background": "#fad165",
			"foreground": "#1d1d1d"
		},
		"13": {
			"background": "#92e1c0",
			"foreground": "#1d1d1d"
		},
		"14": {
			"background": "#9fe1e7",
			"foreground": "#1d1d1d"
		},
		"15": {
			"background": "#9fc6e7",
			"foreground": "#1d1d1d"
		},
		"16": {
			"background": "#4986e7",
			"foreground": "#1d1d1d"
		},
		"17": {
			"background": "#9a9cff",
			"foreground": "#1d1d1d"
		},
		"18": {
			"background": "#b99aff",
			"foreground": "#1d1d1d"
		},
		"19": {
			"background": "#c2c2c2",
			"foreground": "#1d1d1d"
		},
		"20": {
			"background": "#cabdbf",
			"foreground": "#1d1d1d"
		},
		"21": {
			"background": "#cca6ac",
			"foreground": "#1d1d1d"
		},
		"22": {
			"background": "#f691b2",
			"foreground": "#1d1d1d"
		},
		"23": {
			"background": "#cd74e6",
			"foreground": "#1d1d1d"
		},
		"24": {
			"background": "#a47ae2",
			"foreground": "#1d1d1d"
		}
	},
	"event": {
		"1": {
			"background": "#a4bdfc",
			"foreground": "#1d1d1d"
		},
		"2": {
			"background": "#7ae7bf",
			"foreground": "#1d1d1d"
		},
		"3": {
			"background": "#dbadff",
			"foreground": "#1d1d1d"
		},
		"4": {
			"background": "#ff887c",
			"foreground": "#1d1d1d"
		},
		"5": {
			"background": "#fbd75b",
			"foreground": "#1d1d1d"
		},
		"6": {
			"background": "#ffb878",
			"foreground": "#1d1d1d"
		},
		"7": {
			"background": "#46d6db",
			"foreground": "#1d1d1d"
		},
		"8": {
			"background": "#e1e1e1",
			"foreground": "#1d1d1d"
		},
		"9": {
			"background": "#5484ed",
			"foreground": "#1d1d1d"
		},
		"10": {
			"background": "#51b749",
			"foreground": "#1d1d1d"
		},
		"11": {
			"background": "#dc2127",
			"foreground": "#1d1d1d"
		},
	},
}
